### PR TITLE
Introduce tools.go

### PR DESCRIPTION
**Pull request by hackerwins**
**What this PR does / why we need it**:

`tools.go` ensures tool dependencies are kept in sync. This is the recommended way of doing this according to the below.

https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module

After running `go mod tidy`, `go.mod` should remain unchanged.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: tools/tools.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import _ "github.com/gogo/protobuf/gogoproto"
+
+//go:generate go install github.com/gogo/protobuf/protoc-gen-gogo
+import _ "github.com/gogo/protobuf/protoc-gen-gogo"
+
+//go:generate go install github.com/gogo/protobuf/protoc-gen-gofast
+import _ "github.com/gogo/protobuf/protoc-gen-gofast"
+
+//go:generate go install github.com/golangci/golangci-lint/cmd/golangci-lint
+import _ "github.com/golangci/golangci-lint/cmd/golangci-lint"
```
Comment (dc7303) [inline tools/tools.go:27]: Even if the module is not imported, it is confirmed that the module is installed by `go install`. Can you explain the reason for the import modules?

**Review by dc7303 — COMMENTED** [review]
Thanks for the wonderful work. 👍  I learned about `go generate` thanks to this work.

I asked a question about the part I was curious about. Please check.

File: tools/tools.go (hunk 1 above)
Comment (hackerwins) [reply tools/tools.go:27]: Thanks for the comment. Sorry I didn't fully understand the question. But the behavior I understood is below.

Typing `make tools` will run `go generate -tags tools tools/tools.go`. `go generate` runs comments in patterns that begin with `go:generate`. And modules imported into `tools.go` are not used directly in the source code, but are tracked by go mod. For more details please refer below link:

https://github.com/golang/go/wiki/Modules#how-can-i-track-tool-dependencies-for-a-module

Comment (dc7303) [reply tools/tools.go:27]: Thank you for your feedback! 👍

**Review by dc7303 — APPROVED** [review]
LGTM 👍

**Pull request merged**